Startup.cs

```diff
@@ -32,6 +33,11 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers();
+            services.AddControllers().ConfigureApiBehaviorOptions(options =>
+            {
+                // Adds a custom error response factory when ModelState is invalid
+                options.InvalidModelStateResponseFactory = InvalidModelStateResponseFactory.ProduceErrorResponse;
+            });
```

Comment:
When combined with the ApiController attribute, this allows the pipeline to handle validation errors to be returned in a consistent (with the other responses) way.   While the default: 

> BadRequestObjectResult(actionContext.ModelState)

works, the custom factory produces an error that has a success flag like the rest of our responses.

> ErrorResource(errors);